sharding/nodesCoordinator/sovereignNumberOfShardsComputer.go

```diff
@@ -0,0 +1,23 @@
+package nodesCoordinator
+
+type sovereignNumberOfShardsComputer struct {
+}
+
+// NewSovereignNumberOfShardsComputer creates a new instance of the number of shards computer for sovereign
+func NewSovereignNumberOfShardsComputer() *sovereignNumberOfShardsComputer {
+	return &sovereignNumberOfShardsComputer{}
+}
+
+// ComputeNumberOfShards computes the number of shards for the given nodes config
+func (snsc *sovereignNumberOfShardsComputer) ComputeNumberOfShards(config *epochNodesConfig) (nbShards uint32, err error) {
+	nbShards = uint32(len(config.eligibleMap))
```

Comment:
Would it make sense to also check that the map does not contain anything other than shardID == core.SovereignChainID ?

Reply:
yes added